derivation/src/types.rs

```diff
@@ -0,0 +1,76 @@
+use alloc::vec::Vec;
+use alloy_primitives::B256;
+use kona_preimage::PreimageKey;
+use prost::Message as ProtoMessage;
+use prost_derive::Message;
+
+#[derive(Message, Clone, PartialEq)]
+pub struct Preimage {
```

Comment:
Why didn't you use protobuf for defining `Preimage` and `Preimages`?

Reply:
@yoshidan OK, I have confirmed that `optimism-preimage-maker` also refers to this type definition. To say, there's no duplication of definitions.